mesonbuild/modules/i18n.py

```diff
@@ -202,6 +202,8 @@ def _get_source_files(self, sources: T.Iterable[SourcesType]) -> T.Set[mesonlib.
                 source_files.update(source.get_sources())
             elif isinstance(source, build.BothLibraries):
                 source_files.update(source.get('shared').get_sources())
+            elif isinstance(source, (build.CustomTarget, build.CustomTargetIndex)):
+                source_files.update(source.get_outputs())
```

Comment:
Should be something like `File.from_built_file(source.subdir, f) for f in source.get_outputs()`